### ISSUE TITLE
cannot connect to X server

### ISSUE THREAD

**Issue by woafrnraetns** [xingyizhou/CenterTrack#49]
hi, when i try
**python demo.py tracking,ddd --load_model ../models/nuScenes_3Dtracking.pth --dataset nuscenes --pre_hm --track_thresh 0.1 --demo ../videos/nuscenes_mini.mp4 --test_focal_length 633**
i got
/.conda/envs/env1/lib/python3.6/site-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
Running tracking
Using tracking threshold for out threshold! 0.1
Fix size testing.
training chunk_sizes: [32]
input h w: 448 800
heads {'hm': 10, 'reg': 2, 'wh': 2, 'tracking': 2, 'dep': 1, 'rot': 8, 'dim': 3, 'amodel_offset': 2}
weights {'hm': 1, 'reg': 1, 'wh': 0.1, 'tracking': 1, 'dep': 1, 'rot': 1, 'dim': 1, 'amodel_offset': 1}
head conv {'hm': [256], 'reg': [256], 'wh': [256], 'tracking': [256], 'dep': [256], 'rot': [256], 'dim': [256], 'amodel_offset': [256]}
Creating model...
Using node type: (<class 'model.networks.dla.DeformConv'>, <class 'model.networks.dla.DeformConv'>)
Warning: No ImageNet pretrain!!
loaded ../models/nuScenes_3Dtracking.pth, epoch 70
: cannot connect to X server

could you help me find the problem?

**Comment by Muhtasham** [comment]
Hi @woafrnraetns,
try to install xvfb first using this command 
`apt-get install -y xvfb python-opengl ffmpeg > /dev/null 2>&1
then try to run the python file 
`xvfb-run python demo.py

**Comment by bamps53** [comment]
@woafrnraetns You should describe your environment a bit more. Do you run the script on your laptop or remote server? Or in docker container? For me, my docker container in remote server was not ported display. Please try just imshow(img) to make sure you can see the image or not.

**Comment by woafrnraetns** [comment]
thank you a lot.it was true that i was using a remote server without x11.after delete every cv2.imshow, the problem solved

**Comment by wrccrwx** [comment]
To anyone who may concern, if you want to run "demo.py" in environment like colab, you may have to make several modification to the codes:

0.  install torch 1.4.0 and torchvision 0.5.0
1.  replace "cv2.imshow" in function  show_all_image in "debugger.py" with "pass"
2.  replace "cv2.imshow" at line 52 in "demo.py" with "img = cv2.resize(img, (opt.video_w, opt.video_h))".
3. solve the issue of videowriter
"
fourcc = cv2.VideoWriter_fourcc(*'XVID')
      out = cv2.VideoWriter('./{}.avi'.format(
        opt.exp_id + '_' + out_name),fourcc, 30, (
          1920, 1080))
"

**Comment by Muhtasham** [comment]
> To anyone who may concern, if you want to run "demo.py" in environment like colab, you may have to make several modification to the codes:
> 
> 1. install torch 1.4.0 and torchvision 0.5.0
> 2. replace "cv2.imshow" in function  show_all_image in "debugger.py" with "pass"
> 3. replace "cv2.imshow" at line 52 in "demo.py" with "img = cv2.resize(img, (opt.video_w, opt.video_h))".
> 4. solve the issue of videowriter
>    "
>    fourcc = cv2.VideoWriter_fourcc(*'XVID')
>    out = cv2.VideoWriter('./{}.avi'.format(
>    opt.exp_id + '_' + out_name),fourcc, 30, (
>    1920, 1080))
>    "

did you had any issues with importing DCN?

**Comment by sisrfeng** [comment]
> 3\. place "cv2.imshow" at line 52 in "demo.py" with "img = cv2.resize(img, (opt.video_w, opt.video_h))".

the author seems to have fix it
